src/getMigrateDashboards.ts

```diff
@@ -42,7 +43,14 @@ export const getMigrateDashboards =
   ) =>
   <FromDashboardState, ToDashboardState>(
     callback: MigrateDashboardCallback<FromDashboardState, ToDashboardState>,
+    deserialize: (state: FromDashboardState) => FromDashboardState,
+    serialize: (state: ToDashboardState) => ToDashboardState,
   ): void => {
+    // This function returned by `getMigrateDashboards` and accessing its outer scope variable `contentServer` forms a closure.
+    // It causes some parts of the `contentServer` object to not be writable, hence not mutable.
+    // This is done to override that behavior and have a fully mutable `contentServer` object.
```

Comment:
What causes the immutability ? Can't we use `produce` in the right places instead ?
Also, I'm not sure I get the link between the function closure and the immutability.

Reply:
As discussed together this is probably because `migrateCalculatedMeasures` is called before and creates an immutable object in the first place.
https://github.com/activeviam/activeui-migration/pull/70 will lift this problem.